cs244b_project/src/blockchain/manager.rs

```diff
@@ -134,13 +136,16 @@ impl BlockchainManager {
                 "\n\nSuccessfully finalized chain, new finalized chain {}\n",
                 self.finalized_chain
             );
-
         }
     }
 
-    pub fn export_local_chain(&self) -> LocalChain {
-        // not totally sure on how to specify the where to export to functionality so for now this just pretty-prints the finalized chain representing the log
-        self.finalized_chain.clone()
+    pub fn fetch_local_finalized_chain(&self) -> LocalChain { self.finalized_chain.clone() }
+    pub fn export_local_finalized_chain_to_file(&self, local_file_path: String) {
+        info!("exporting local finalized chain to: {}", local_file_path);
+        fs::write(local_file_path, serde_json::to_string_pretty(&self.fetch_local_finalized_chain()).unwrap()).expect("failed to write to file!");
```

Comment:
This entirely replaces the contents of the file https://doc.rust-lang.org/std/fs/fn.write.html